pkg/kubectl/cmd/apply_test.go

```diff
@@ -61,18 +62,24 @@ func validateApplyArgs(cmd *cobra.Command, args []string) error {
 }
 
 const (
-	filenameRC             = "../../../test/fixtures/pkg/kubectl/cmd/apply/rc.yaml"
-	filenameRCNoAnnotation = "../../../test/fixtures/pkg/kubectl/cmd/apply/rc-no-annotation.yaml"
-	filenameRCLASTAPPLIED  = "../../../test/fixtures/pkg/kubectl/cmd/apply/rc-lastapplied.yaml"
-	filenameSVC            = "../../../test/fixtures/pkg/kubectl/cmd/apply/service.yaml"
-	filenameRCSVC          = "../../../test/fixtures/pkg/kubectl/cmd/apply/rc-service.yaml"
-	filenameNoExistRC      = "../../../test/fixtures/pkg/kubectl/cmd/apply/rc-noexist.yaml"
-	filenameRCPatchTest    = "../../../test/fixtures/pkg/kubectl/cmd/apply/patch.json"
-	dirName                = "../../../test/fixtures/pkg/kubectl/cmd/apply/testdir"
-	filenameRCJSON         = "../../../test/fixtures/pkg/kubectl/cmd/apply/rc.json"
+	filenameRC                   = "../../../test/fixtures/pkg/kubectl/cmd/apply/rc.yaml"
+	filenameRCNoAnnotation       = "../../../test/fixtures/pkg/kubectl/cmd/apply/rc-no-annotation.yaml"
+	filenameRCLASTAPPLIED        = "../../../test/fixtures/pkg/kubectl/cmd/apply/rc-lastapplied.yaml"
+	filenameSVC                  = "../../../test/fixtures/pkg/kubectl/cmd/apply/service.yaml"
+	filenameRCSVC                = "../../../test/fixtures/pkg/kubectl/cmd/apply/rc-service.yaml"
+	filenameNoExistRC            = "../../../test/fixtures/pkg/kubectl/cmd/apply/rc-noexist.yaml"
+	filenameLocalChangedRC       = "../../../test/fixtures/pkg/kubectl/cmd/apply/rc-changed.yaml"
+	filenameApplyDiffRESP        = "../../../test/fixtures/pkg/kubectl/cmd/apply/rc-resp.json"
+	filenameApplyDiffChangedRESP = "../../../test/fixtures/pkg/kubectl/cmd/apply/rc-changed-resp.json"
+	filenameRCPatchTest          = "../../../test/fixtures/pkg/kubectl/cmd/apply/patch.json"
+	dirName                      = "../../../test/fixtures/pkg/kubectl/cmd/apply/testdir"
+	filenameRCJSON               = "../../../test/fixtures/pkg/kubectl/cmd/apply/rc.json"
 
 	filenameWidgetClientside = "../../../test/fixtures/pkg/kubectl/cmd/apply/widget-clientside.yaml"
 	filenameWidgetServerside = "../../../test/fixtures/pkg/kubectl/cmd/apply/widget-serverside.yaml"
+
+	fileNameGitDiffResult    = "./git-diff.result"
```

Comment:
@shiywang I had asked you to use "./" to avoid relying on the presence of a "/tmp" directory. It's a minor point but it could break the tests on an unusual platform or in a situation with unusual permissions. I like @mengqiy's suggestion, but instead of "/tmp", try https://golang.org/pkg/io/ioutil/#TempFile. I think that would be the best of both worlds.